Fails to compile with Qt 5.14

Starting with Qt 5.14 qtmir doesn't compile anymore:
```
/home/pmos/build/src/qtmir-f1a612ce524c57f689b68c6f3b57fdfe8d169560/src/modules/Unity/Application/mirsurfaceitem.cpp: In member function 'virtual QSGNode* qtmir::MirSurfaceItem::updatePaintNode(QSGNode*, QQuickItem::UpdatePaintNodeData*)':
/home/pmos/build/src/qtmir-f1a612ce524c57f689b68c6f3b57fdfe8d169560/src/modules/Unity/Application/mirsurfaceitem.cpp:237:20: error: no matching function for call to 'QSGDefaultInternalImageNode::QSGDefaultInternalImageNode()'
  237 |         node = new QSGDefaultInternalImageNode;
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/pmos/build/src/qtmir-f1a612ce524c57f689b68c6f3b57fdfe8d169560/src/modules/Unity/Application/mirsurfaceitem.cpp:35:
/usr/include/QtQuick/5.14.1/QtQuick/private/qsgdefaultinternalimagenode_p.h:78:5: note: candidate: 'QSGDefaultInternalImageNode::QSGDefaultInternalImageNode(QSGDefaultRenderContext*)'
   78 |     QSGDefaultInternalImageNode(QSGDefaultRenderContext *rc);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/QtQuick/5.14.1/QtQuick/private/qsgdefaultinternalimagenode_p.h:78:5: note:   candidate expects 1 argument, 0 provided
make[2]: *** [src/modules/Unity/Application/CMakeFiles/unityapplicationplugin.dir/build.make:194: src/modules/Unity/Application/CMakeFiles/unityapplicationplugin.dir/mirsurfaceitem.cpp.o] Error 1

```
See qtdeclarative commit: https://github.com/qt/qtdeclarative/commit/341ab7708049b1a3f559b76f16393e688951a938

Fixing this seems to be non-trivial, at least I haven't managed to do it yet.

--- a/src/modules/Unity/Application/mirsurfaceitem.cpp
+++ b/src/modules/Unity/Application/mirsurfaceitem.cpp
@@ -32,6 +32,10 @@
 #include <QQmlEngine>
 #include <QQuickWindow>
 #include <QScreen>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+#include <private/qquickwindow_p.h>
+#include <private/qsgdefaultrendercontext_p.h>
+#endif
 #include <private/qsgdefaultinternalimagenode_p.h>
 #include <QTimer>
 #include <QSGTextureProvider>
@@ -234,7 +238,12 @@ QSGNode *MirSurfaceItem::updatePaintNode(QSGNode *oldNode, UpdatePaintNodeData *
     m_textureProvider->smooth = smooth();
     QSGDefaultInternalImageNode *node = static_cast<QSGDefaultInternalImageNode*>(oldNode);
     if (!node) {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+        QSGRenderContext *rc = QQuickWindowPrivate::get(window())->context;
+        node = new QSGDefaultInternalImageNode(static_cast<QSGDefaultRenderContext *>(rc));
+#else
         node = new QSGDefaultInternalImageNode;
+#endif
         node->setMipmapFiltering(QSGTexture::None);
         node->setHorizontalWrapMode(QSGTexture::ClampToEdge);
         node->setVerticalWrapMode(QSGTexture::ClampToEdge);

